notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,498 @@
+```python
+# HIDDEN
+def df_interact(df, nrows=7, ncols=7):
+    '''
+    Outputs sliders that show rows and columns of df
+    '''
+    def peek(row=0, col=0):
+        return df.iloc[row:row + nrows, col:col + ncols]
+    if len(df.columns) <= ncols:
+        interact(peek, row=(0, len(df) - nrows, nrows), col=fixed(0))
+    else:
+        interact(peek,
+                 row=(0, len(df) - nrows, nrows),
+                 col=(0, len(df.columns) - ncols))
+    print('({} rows, {} columns) total'.format(df.shape[0], df.shape[1]))
+```
+
+## Cross-Validation
+
+In the previous chapter, we observed that we needed a more accurate way of simulating the test error to manage the bias-variance trade off. Cross-validation provides a method of estimating our model error using a single observed dataset by separating data used for training from the data used for model selection and final accuracy. 
+
+## Train-Validation-Test Split
+
+One way to accomplish this is to split the original dataset into three disjoint subsets:
+
+* Training set: The data used to fit the model.
+* Validation set: The data used to select features.
+* Test set: The data used to report the model's final accuracy.
+
+After splitting, we select a set of features and a model based on the following procedure:
+
+1. For each potential set of features, fit a model using the training set. The error of a model on the training set is its *training error*.
+1. Check the error of each model on the validation set: its *validation error*. Select the model that achieves the lowest validation error. This is the final choice of features and model.
+1. Calculate the *test error*, error of the final model on the test set. This is the final reported accuracy of the model. We are forbidden from adjusting the features or model to decrease test error; doing so effectively converts the test set into a validation set. Instead, we must collect a new test set after making further changes to the features or the model.
+
+This process allows us to more accurately determine the model to use than using the training error alone.
+
+**Size of the train-validation-test split**
+
+The train-validation-test split commonly uses 70% of the data as the training set, 15% as the validation set, and the remaining 15% as the test set. Increasing the size of the training set helps model accuracy but causes more variation in the validation and test error.
+
+## Training Error and Test Error
+
+A model is of little use to us if it fails to generalize to unseen data from the population. The test error provides an accurate representation of the model's performance on new data since we do not use the test set to train the model or select features.
+
+In general, the training error decreases as we add complexity to our model with additional features or more complex prediction mechanisms. The test error, on the other hand, decreases up to a certain amount of complexity then increases again as the model overfits the training set.
```

Comment:
Since this section now comes after bias-variance, one nice thing to do here is to explain the last statement a bit. Something like "the test error decreases then increases b/c bias goes down, then variance goes up."

Reply:
fixed